dll/win32/shell32/shlexec.cpp

```diff
@@ -2383,14 +2421,63 @@ ShellExecuteExA(LPSHELLEXECUTEINFOA sei)
 
 /*************************************************************************
  * ShellExecuteExW                [SHELL32.293]
- *
  */
 BOOL
 WINAPI
 DECLSPEC_HOTPATCH
 ShellExecuteExW(LPSHELLEXECUTEINFOW sei)
 {
-    return SHELL_execute(sei, SHELL_ExecuteW);
+    HRESULT hrCoInit;
+    DWORD dwError;
+    ULONG fOldMask;
+
+    hrCoInit = SHCoInitializeAnyApartment();
+
+    if (sei->cbSize != sizeof(SHELLEXECUTEINFOW))
+    {
+        dwError = ERROR_ACCESS_DENIED;
+        sei->hInstApp = (HINSTANCE)ERROR_ACCESS_DENIED;
+    }
+    else
+    {
+        fOldMask = sei->fMask;
+
+        sei->fMask |= 0x800;
+
+        if (SHRegGetBoolUSValueW(L"Software\\Microsoft\\Windows\\CurrentVersion\\Explorer",
+                                 L"MaximizeApps", FALSE, FALSE))
+        {
+            switch (sei->nShow)
+            {
+                case SW_SHOWNORMAL:
+                case SW_SHOW:
+                case SW_RESTORE:
+                case SW_SHOWDEFAULT:
+                    sei->nShow = SW_SHOWMAXIMIZED;
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        if (!(sei->fMask & SEE_MASK_NOASYNC) && SHELL_InRunDllProcess())
+            sei->fMask |= SEE_MASK_WAITFORINPUTIDLE | SEE_MASK_NOASYNC;
```

Comment:
Which rundll32 command breaks without this?

Reply:
This case considers the batch process.
The function won't returnuntil done.